File: tags/Release/SR-RBD/src/ddb/restore/msg/RestoreTableList.java
package ddb.restore.msg;

import java.io.IOException;
import java.util.LinkedList;
import java.util.List;

import ddb.communication.DataInputStream;
import ddb.communication.DataOutputStream;
import ddb.db.TableVersion;
import ddb.msg.MessageType;

public class RestoreTableList extends RestoreMessage {

	private List<TableVersion> tables = new LinkedList<TableVersion>();
	
	public RestoreTableList()
	{
		
	}
	
	public RestoreTableList(List<TableVersion> list)
	{
		this.tables = list;
	}
	
	public void setTables(List<TableVersion> tables) {
		this.tables = tables;
	}

	public List<TableVersion> getTables() {
		return tables;
	}
	
	@Override
	public void fromBinary(DataInputStream s) throws IOException {
		// read count
		int count = s.readInt();
		setTables(new LinkedList<TableVersion>());
		
		// read contents
		for(int i = 0; i < count; ++i)
			getTables().add(new TableVersion(s));
	}
	
	@Override
	public MessageType getType() {
		return MessageType.RESTORE_TABLELIST;
	}

	@Override
	public void toBinary(DataOutputStream s) throws IOException {
		// write length
		s.writeInt(getTables().size());
		
		// write tables
		for(TableVersion tv : getTables())
			tv.toBinary(s);
	}

}

File: C#/RBD/RBD/Restore/Msg/RestoreTableList.cs
// + TODO check

using System.Collections;
using System.Collections.Generic;
using RBD.DB;
using RBD.Communication;

namespace RBD.Restore.Msg
{
    public class RestoreTableList : RestoreMessage {

        private IList<TableVersion> tables = new List<TableVersion>();
    	
	    public RestoreTableList()
	    {
    		
	    }

        public RestoreTableList(IList<TableVersion> list)
        {
            this.tables = list;
        }

        public void setTables(IList<TableVersion> tables)
        {
            this.tables = tables;
        }

        public IList<TableVersion> getTables()
        {
            return tables;
        }
    	
	    override public void FromBinary(DataInputStream s)  {
		    // read count
		    int count = s.readInt();
            setTables(new List<TableVersion>());

            // read contents
            for(int i = 0; i < count; ++i)
                getTables().Add(new TableVersion(s));
	    }

        override public MessageType GetMessageType()
        {
		    return MessageType.RESTORE_TABLELIST;
	    }

        override public void ToBinary(DataOutputStream s)
        {
		    // write length
            s.Write((int)getTables().Count);

            // write tables
            foreach (TableVersion tv in getTables())
                tv.ToBinary(s);
	    }
    }
}
